pkg/kubelet/sysctl/safe_sysctls.go

```diff
@@ -51,19 +51,29 @@ var safeSysctlsIncludeReservedPorts = []string{
 // - it is namespaced in the container or the pod
 // - it is isolated, i.e. has no influence on any other pod on the same node.
 func SafeSysctlAllowlist() []string {
-	if goruntime.GOOS == "linux" {
-		// make sure we're on a new enough kernel that the ip_local_reserved_ports sysctl is namespaced
-		kernelVersion, err := getKernelVersion()
-		if err != nil {
-			klog.ErrorS(err, "Failed to get kernel version, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list")
-			return safeSysctls
-		}
-		if kernelVersion.LessThan(version.MustParseGeneric(ipLocalReservedPortsMinNamespacedKernelVersion)) {
-			klog.ErrorS(nil, "Kernel version is too old, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list", "kernelVersion", kernelVersion)
-			return safeSysctls
-		}
+	if goruntime.GOOS != "linux" {
+		return append(safeSysctls, sysctlIpLocalReserverdPorts, sysctlTcpKeepAliveTime)
```

Comment:
This seems wrong.

Really, this function should never be called on non-Linux, and it should just panic if it does get called. But failing that, it should return `nil` on non-Linux, because this function only knows what sysctls are safe on Linux, so if the platform is not Linux, then we have no idea which sysctls are safe.

Reply:
Thanks for the point, yes, this function should never be called on non-Linux.